fabric-enchantments-v1/src/main/java/net/fabricmc/fabric/api/enchantment/v1/EnchantmentTargetRegistry.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.enchantment.v1;
+
+import net.fabricmc.fabric.impl.enchantment.EnchantmentTargetRegistryImpl;
+
+import java.util.Iterator;
+
+/**
+ * Used to register custom enchantment targets.
+ *
+ * @see FabricEnchantment
+ * @see FabricEnchantmentTarget
+ */
+public interface EnchantmentTargetRegistry {
+	EnchantmentTargetRegistry INSTANCE = EnchantmentTargetRegistryImpl.INSTANCE;
+
+	/**
+	 * Registers a new {@link FabricEnchantmentTarget}.
+	 * @param enchantmentTarget the custom target
+	 */
+	void register(FabricEnchantmentTarget enchantmentTarget);
+
+	/**
+	 * Gets an iterator over all registered enchantment targets.
+	 * @return the iterator
+	 */
+	Iterator<FabricEnchantmentTarget> getIterator();
```

Comment:
Does this need to be exposed via the api?

Reply:
If you want to see all existing targets or want to find out which targets match on an item - then possibly yes.

Reply:
why not make this object iterable 🤔

Reply:
Not relevant anymore